ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/LoadConstraint.java

```diff
@@ -124,4 +137,72 @@ public TaskTracker tasks() throws IOException, SemanticException {
     }
   }
 
+  private boolean isPrimaryKeysAlreadyLoaded(String pksMsgString) throws Exception {
+    AddPrimaryKeyMessage msg = deserializer.getAddPrimaryKeyMessage(pksMsgString);
+    List<SQLPrimaryKey> pksInMsg = msg.getPrimaryKeys();
+    if (pksInMsg.isEmpty()) {
+      return true;
```

Comment:
is it a valid scenario ?

Reply:
Not sure. But the base code for constraint handling did it. So, having it here as well.